### ISSUE TITLE
One brand of steering wheels, 3 different drivers for Linux

### ISSUE THREAD

**Issue by leillo1975** [emtek995/TMX-driver#5]
Would it not be better to unify efforts into one? For example Logitech steering wheel drivers are almost all in [new-lg4ff](https://github.com/berarma/new-lg4ff) (G920 and G923 XBOX has kernel support).

At Thrustmaster on the other hand, there are 3 different projects that I know of:
**T150:** https://github.com/scarburato/t150_driver
**T300RS, T248 and TX** (and more): https://github.com/Kimplul/hid-tmff2
and now your driver for TMX
I forget **TMDRV**: https://gitlab.com/her0/tmdrv

I'm not at all despising your project and your efforts, which are always welcome. It's just an opinion.

**Comment by pastaq** [comment]
@leillo1975 I agree with you. I think only tmff2 is being maintained at this point. This repo hasn't had a commit in at least a year and TMDRV hasn't had a commit in 5. T150 was updated 9 months ago but TBH it would make the most sense to add t150 support to tmff as well. This repo would be a good reference for adding initial support to tmff I think.

As an aside, do you have a TMX device? I don't currently have a windows machine to do usbcap for FF events.

**Comment by leillo1975** [comment]
No, not really, but I am very interested in everything related to Linux flywheel support. I really like driving games. I currently have an old G29, but my idea is to buy another model, probably from Thrustmaster. That is why I am interested in the support of this brand

**Comment by emtek995** [comment]
Hello all, first I wanna say thanks for interest in this project. I am aware of the other projects available for the TMX.
All this driver does at the moment is exactly what [TMDRV](https://gitlab.com/her0/tmdrv) does, except as a kernel module. My local copy of this repo has a branch where I've started integrating code from [tmff2](https://github.com/Kimplul/hid-tmff2). I ended up bricking my wheel some how experimenting with force feedback support, so I'm not going to push that branch up. It shouldn't be too difficult to integrate this repo into tmff2, but I've got no wheel to test on anymore until I figure out how to unbrick mine.

**Comment by pastaq** [comment]
> Hello all, first I wanna say thanks for interest in this project. I am aware of the other projects available for the TMX. All this driver does at the moment is exactly what [TMDRV](https://gitlab.com/her0/tmdrv) does, except as a kernel module. My local copy of this repo has a branch where I've started integrating code from [tmff2](https://github.com/Kimplul/hid-tmff2). I ended up bricking my wheel some how experimenting with force feedback support, so I'm not going to push that branch up. It shouldn't be too difficult to integrate this repo into tmff2, but I've got no wheel to test on anymore until I figure out how to unbrick mine.

Oof, that's rough. Looks like I'm the only one with a device to work on ATM. At this point I'll either break it and buy a Logitech wheel or get it working.

**Comment by Kimplul** [comment]
> [...] I ended up bricking my wheel some how experimenting with force feedback support, so I'm not going to push that branch up. [...]

Sorry to butt in, but did you try putting the wheel into bootloader mode? I thought I bricked my wheel when I first messed around with its driver, but I got it working again by reinstalling the firmware via the bootloader. There are instructions on the Thrustmaster website, they're not exactly highly visible but do exist: https://ts.thrustmaster.com/download/accessories/Manuals/TMX/TMX_Bootloader_Method.pdf

**Comment by pastaq** [comment]
> > [...] I ended up bricking my wheel some how experimenting with force feedback support, so I'm not going to push that branch up. [...]
> 
> Sorry to butt in, but did you try putting the wheel into bootloader mode? I thought I bricked my wheel when I first messed around with its driver, but I got it working again by reinstalling the firmware via the bootloader. There are instructions on the Thrustmaster website, they're not exactly highly visible but do exist: https://ts.thrustmaster.com/download/accessories/Manuals/TMX/TMX_Bootloader_Method.pdf

Just to second this, I also ended up getting stuck in the bootloader mode while messing in a vm until I installed Windows and ran the firmware updater
